packages/remix-dev/config.ts

```diff
@@ -469,7 +469,7 @@ export async function resolveConfig(
   let pkgJson = await PackageJson.load(rootDirectory);
   let deps = pkgJson.content.dependencies ?? {};
 
-  if (isSpaMode) {
+  if (isSpaMode && appConfig.future?.unstable_singleFetch != true) {
```

Comment:
Default to the streaming entry server in SPA mode w/singleFetch if the user doesn't have an `entry.server`